sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -455,6 +456,26 @@ public boolean bucketAccessible(GcsPath path) throws IOException {
   }
 
   /**
+   * Returns whether the GCS bucket has TTL assigned.
+   *
+   * @param path the path to GCS bucket
+   * @return true if TTL is assigned to bucket, false otherwise
+   */
+  public boolean bucketHasTTL(GcsPath path) throws IOException {
+    Bucket bucket = getBucket(path, createBackOff(), Sleeper.DEFAULT);
+    if (bucket != null && bucket.getLifecycle() != null
+        && bucket.getLifecycle().getRule() != null) {
+      for (Rule r : bucket.getLifecycle().getRule()) {
+        if ("Delete".equalsIgnoreCase(r.getAction().getType())) {
```

Comment:
Delete can probably be moved into a constant field. I don't believe there's an existing constant, which is a shame.

We may also consider warning (for benefit of the users) if there's a `SetStorageClass` rule, just because that might increase consts unexpectedly.